Networking/Networking/ApplicationPassword/ApplicationPasswordUseCase.swift

```diff
@@ -28,3 +34,219 @@ protocol ApplicationPasswordUseCase {
     ///
     func deletePassword() async throws
 }
+
+final class DefaultApplicationPasswordUseCase: ApplicationPasswordUseCase {
+    /// WordPress.com Credentials.
+    ///
+    private let credentials: Credentials
+
+    /// SiteID needed when using WPCOM credentials
+    ///
+    private let siteID: Int64
+
+    /// To generate and delete application password
+    ///
+    private let network: Network
+
+    /// Stores the application password
+    ///
+    private let keychain: Keychain
+
+    /// Used to name the password in wpadmin.
+    ///
+    private var applicationPasswordName: String {
+        get async {
+            await UIDevice.current.model
+        }
+    }
+
+    init(siteID: Int64,
+         networkcredentials: Credentials,
+         network: Network? = nil,
+         keychain: Keychain = Keychain(service: "com.automattic.woocommerce.applicationpassword")) {
+        self.siteID = siteID
+        self.credentials = networkcredentials
+        self.keychain = keychain
+
+        if let network {
+            self.network = network
+        } else {
+            self.network = ApplicationPasswordNetwork(credentials: networkcredentials)
+        }
+    }
+
+    /// Returns the locally saved ApplicationPassword if available
+    ///
+    var applicationPassword: ApplicationPassword? {
+        guard let password = keychain.applicationPassword,
+              let username = keychain.applicationPasswordUsername else {
+            return nil
+        }
+        return ApplicationPassword(wpOrgUsername: username, password: Secret(password))
+    }
+
+    /// Generates new ApplicationPassword
+    ///
+    /// - Returns: Generated `ApplicationPassword` instance
+    ///
+    func generateNewPassword() async throws -> ApplicationPassword {
+        let password = try await createApplicationPasswordUsingWPCOMAuthToken()
```

Comment:
Should we handle deleting and generating the password again within this use case, or do you think it's better that it's handled from the outside? I just thought it would be more straightforward for the consumer of this use case.

Reply:
I agree. Started deleting the existing password when duplicate name error occurs - 38c7deca8e118b8ceff9599a758fad33ecfabf1f